### PR TITLE
CMR-10182: Enforce Keyword Validation Rules at CMR Ingest

### PR DESCRIPTION
**Pull request by jmaeng72**
# Overview

### What is the feature/fix?

Ingest fields are validated, but at ingest, users will get a warning and still be allowed to ingest. We should enforce keywords at ingest so no invalid keywords get ingested into the CMR.

### What is the Solution?

set cmr-validate-keyword default to be true

### What areas of the application does this impact?

Any users ingesting will now have to set cmr-validate-keyword = false specifically if they want to ingest with only warnings.

Previous behavior was cmr-validate-keywords = false by default, but now cmr-validate-keywords = true will become the new default for ingest

# Checklist

- [ ] I have updated/added unit and int tests that prove my fix is effective or that my feature works
- [ ] New and existing unit and int tests pass locally and remotely
- [ ] clj-kondo has been run locally and all errors corrected
- [ ] I have removed unnecessary/dead code and imports in files I have changed
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have cleaned up integration tests by doing one or more of the following:
  - migrated any are2 tests to are3 in files I have changed
  - de-duped, consolidated, removed dead int tests
  - transformed applicable int tests into unit tests
  - refactored to reduce number of system state resets by updating fixtures (use-fixtures :each (ingest/reset-fixture {})) to be :once instead of :each

### CODE REVIEW

File: system-int-test/test/cmr/system_int_test/bootstrap/fingerprint_test.clj [hunk 1]
```diff
@@ -82,7 +83,7 @@
                     (get-in latest-revision-concept [:extra-fields :fingerprint]))))))
 
        (testing "update fingerprint of a deleted variable"
-         (let [{:keys [concept-id revision-id]} (ingest/delete-concept var2-concept)
+         (let [{:keys [_ _]} (ingest/delete-concept var2-concept)
```
Comment (eereiter) [inline system-int-test/test/cmr/system_int_test/bootstrap/fingerprint_test.clj:86]: Put (ingest/delete-concept var2-concept) before the let statement 
Remove the first let statement.
The code should look like 
`
(testing "update fingerprint of a deleted variable"
  (ingest/delete-concept var2-concept)
  (let [latest-revision-concept (mdb/get-concept var2-concept-id)]
`
if you want to keep the first line in the let statement then just remove {:keys [_ _]} and replace with _

File: system-int-test/test/cmr/system_int_test/search/granule/granule_psa_search_test.clj [hunk 2]
```diff
@@ -103,16 +103,18 @@
...
 
-        coll1 (d/ingest "PROV1" (dc/collection {:product-specific-attributes [psa1 psa2]}))
+        coll1 (d/ingest "PROV1" (dc/collection {:product-specific-attributes [psa1 psa2]}) {:validate-keywords false})
         gran1 (d/ingest "PROV1" (dg/granule coll1 {:product-specific-attributes [(dg/psa "alpha" ["ab" "bc"])
                                                                                  (dg/psa "bravo" ["cd" "bf"])]}))
         gran2 (d/ingest "PROV1" (dg/granule coll1 {:product-specific-attributes [(dg/psa "bravo" ["ab"])]}))
 
-        coll2 (d/ingest "PROV1" (dc/collection {:product-specific-attributes [psa2 psa3]}))
+        coll2 (d/ingest "PROV1" (dc/collection {:product-specific-attributes [psa2 psa3]})
+                        {:validate-keywords false})
```
Comment (eereiter) [inline system-int-test/test/cmr/system_int_test/search/granule/granule_psa_search_test.clj:112]: You don't need to do this but line 112 should more at the end of 111

File: system-int-test/test/cmr/system_int_test/search/granule/granule_psa_search_test.clj [hunk 3]
```diff
@@ -103,16 +103,18 @@
...
         gran3 (d/ingest "PROV1" (dg/granule coll2 {:product-specific-attributes [(dg/psa "bravo" ["aa" "bf"])]}))
         gran4 (d/ingest "PROV1" (dg/granule coll2 {:product-specific-attributes [(dg/psa "charlie" ["az"])]}))
 
-        coll3 (d/ingest "PROV1" (dc/collection {:product-specific-attributes [psa4]}))
+        coll3 (d/ingest "PROV1" (dc/collection {:product-specific-attributes [psa4]})
+                        {:validate-keywords false})
```
Comment (eereiter) [inline system-int-test/test/cmr/system_int_test/search/granule/granule_psa_search_test.clj:117]: You don't need to do this but line 117 should more at the end of 116

File: system-int-test/test/cmr/system_int_test/search/granule/granule_psa_search_test.clj [hunk 4]
```diff
@@ -275,12 +277,13 @@
...
         psa3 (dc/psa {:name "charlie" :data-type :float :value 45})
-        coll1 (d/ingest "PROV1" (dc/collection {:product-specific-attributes [psa1 psa2]}))
+        coll1 (d/ingest "PROV1" (dc/collection {:product-specific-attributes [psa1 psa2]}) {:validate-keywords false})
         gran1 (d/ingest "PROV1" (dg/granule coll1 {:product-specific-attributes [(dg/psa "alpha" [10.5 123])
                                                                                  (dg/psa "bravo" [-12])]}))
         gran2 (d/ingest "PROV1" (dg/granule coll1 {:product-specific-attributes [(dg/psa "bravo" [10.5 123])]}))
 
-        coll2 (d/ingest "PROV1" (dc/collection {:product-specific-attributes [psa1 psa2 psa3]}))
+        coll2 (d/ingest "PROV1" (dc/collection {:product-specific-attributes [psa1 psa2 psa3]})
+                        {:validate-keywords false})
```
Comment (eereiter) [inline system-int-test/test/cmr/system_int_test/search/granule/granule_psa_search_test.clj:286]: You don't need to do this but line 286 should more at the end of 285

**Review by eereiter — CHANGES_REQUESTED** [review]
a couple of really minor changes.

I am not asking you to change what I am about to say: I do wonder if it would have been better/easier to just put {:validate-keywords false} into the ingest function and just override to true when needed for testing.

**Comment by jmaeng72** [comment]
I was worried that overriding the ingest test func would cause issue with other tests that were using it with the intention of having validate=true

File: system-int-test/test/cmr/system_int_test/bootstrap/fingerprint_test.clj [hunk 5]
```diff
@@ -137,7 +140,7 @@
            var2-concept (mdb/get-concept var2-concept-id)
            var2-fingerprint (get-in var2-concept [:extra-fields :fingerprint])
            var3-concept (mdb/get-concept var3-concept-id)
-           var3-fingerprint (get-in var3-concept [:extra-fields :fingerprint])
+           _ (get-in var3-concept [:extra-fields :fingerprint])
```
Comment (TylerHeald1) [inline system-int-test/test/cmr/system_int_test/bootstrap/fingerprint_test.clj:143]: if the value isn't used can we just remove this

File: system-int-test/test/cmr/system_int_test/bootstrap/virtual_products_test.clj [hunk 6]
```diff
@@ -103,14 +102,14 @@
           ;; collections to test the edge case documented in CMR-2169
           coll1 (d/ingest "LPDAAC_ECS"
                           (dc/collection {:entry-title "ASTER L2 Surface Emissivity V003"
-                                          :native-id "NID-1"}))
+                                          :native-id "NID-1"}) {:validate-keywords false})
           _ (ingest/delete-concept (d/item->concept coll1 :echo10))
-          source-collections (vp/ingest-source-collections)
+          source-collections (vp/ingest-source-collections (cmr.system-int-test.utils.virtual-product-util/source-collections) {:validate-keywords false})
```
[the remark this answers is not in the document]
Comment (TylerHeald1) [reply system-int-test/test/cmr/system_int_test/bootstrap/virtual_products_test.clj:107]: Already imported as [cmr.system-int-test.utils.virtual-product-util :as vp] so just use vp
